exec/java-exec/src/main/resources/rest/profile/profile.ftl

```diff
@@ -50,6 +50,34 @@
         "info": false
       }
     );} );
+
+    //Show cancellation status
+    function showCancelPopup(queryId) {
+        document.getElementById("cancelText").innerHTML = "Issued cancellation for query ID: <br>"+queryId+"<br>Click to continue";
+        document.getElementById("cancelPopup").style.display = "block";
+    }
+
+    //Close the cancellation status popup
+    function closePopup() {
+        document.getElementById("cancelPopup").style.display = "none";
+        location.reload(true);
+    }
+
+    //Cancel query
+    function cancelQuery(queryId) {
+        var cancelRestURL = location.protocol + "//" + location.hostname + "/profiles/cancel/" + queryId;
+        let restPort = location.port;
+        if (restPort != 0) {
```

Comment:
`location.port` returns 0 if the port number is standard. However, I've changed this because I realized I don't really need the server as the `/profiles/cancel/...` will be made on the same server that served the profile UI.